ompi/mca/pml/base/pml_base_select.c

```diff
@@ -385,3 +378,38 @@ mca_pml_base_pml_check_selected(const char *my_pml,
     free(remote_pml);
     return OMPI_SUCCESS;
 }
+
+int
+mca_pml_base_pml_check_selected(const char *my_pml,
+                                ompi_proc_t **procs,
+                                size_t nprocs)
+{
+    int ret = 0;
+    size_t i;
+
+    if (opal_pmix_base_async_modex) {
+        /*
+         * If direct modex, then compare our PML with the peer's PML
+         * for all procs
+         */
+        for (i = 0; i < nprocs; i++) {
```

Comment:
I don't think this is a good solution. With this, you are hitting the negative sides of both approaches:
* still all the blobs are collected as you are requesting the data from all of them. Technically this will be a full modex thus all of the memory footpring concerns.
* on the other hand instead of doing it in an efficient collective fashion, it will be done using slow one-by-one direct modex fetching.

Given all of that, here are my thoughts:
* With this change and prior, in case of a full modex, everyone compares to `rank == 0`. So some of procs might be ok, some not.
* What if, in case of direct modex, the comparison will be performed prior to connection establishment? This will delay the detection of an error to some undefined time, but will ensure that we only fetch the data we need for a particular communication graph.